add session-06 (#28)
gazaskygeeks/women-crash-course: coursebook/session-06/flexbox.md

# Flexbox 👩‍💻

> flexbox is technique to layout,organize and distribute HTML elements in efficient way,also manage the space between elements in more sophisticated way.

Reviewer: ```suggestion
flexbox layout allows you to organize and distribute HTML elements inefficiently way, also manage the space between elements in a more sophisticated way.
```

**column-reverse**
```css
.container{
    flex-direction: column-reverse;
}
```
Here the elments will start order from bottom to top.

### Manage Extra Space

right now let's learn how to manage the extra space in the flex system, let's start in manage the space in the main axis:
>here we will assume two things:
> 1-  space is the space in the **Main Axis**
> 2-  the direction is row (horizontally)
> 

we can manage the main axis space using ```justify-content``` in flex-container: 

**center**
```css
.container{
    justify-content: center;
}
```
Here the elements will place at the center and extra spaces will be around them

**flex-start**
```css
.container{
    justify-content: flex-start;
}
```
Here the space will place after the last element

**flex-end**
```css
.container{
    justify-content: flex-end;
}
```
Here the space will place before the first element.

**space-between**
```css
.container{
    justify-content: space-between;
}
```
Here the space will place between all the elements equally.

**space-around**
```css
.container{
    justify-content: space-around;
}
```
Here the space will place between each element equally.

> And you can manage the cross axis with above values (center, flex-start ...etc) but using ```align-items```. 

Reviewer 2: ```suggestion
And you can manage the cross axis with above values (center, flex-start ...etc) by using ```align-items```. 
```

>here we will assume two things:
> 1-  space is the space in the **Cross Axis**
> 2-  the direction is row (horizontally)

**flex-start**
```css
.container{
    align-items: flex-start;
}
```
Here the extra space (of the cross axis) will be bellow flex-elements.

> when i mean after all elements I mean the elements will start to place into the container from where the flex-start in the cross axis.

> **flex-start**

Reviewer 2: ```suggestion
**flex-end**
```
